src/output.c

```diff
@@ -453,6 +453,22 @@ static void output_id3(uint8_t *buf, unsigned int len)
                           price, until, url, seller, desc, received_as);
             }
         }
+        else if (memcmp(buf + off, "XHDR", 4) == 0)
+        {
+            int xhdr_length;
+            log_debug("XHDR MIME hash = %02X %02X %02X %02X", buf[off+10], buf[off+11], buf[off+12], buf[off+13]);
+            log_debug("XHDR Parameter ID = %02X", buf[off+14]);
+            xhdr_length = buf[off + 15];
+            if (xhdr_length > 0) {
```

Comment:
This should probably also check that `xhdr_length` isn't too large (which could occur if the data is corrupted when the signal quality is poor). That could be done like so:

`if (xhdr_length > 0 && xhdr_length <= frame_len - 6) {`